compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -35,25 +35,19 @@ trait QuotesAndSplices { self: Typer =>
    *  while tracking the quotation level in the context.
    */
   def typedQuote(tree: untpd.Quote, pt: Type)(implicit ctx: Context): Tree = track("typedQuote") {
-    val qctx = inferImplicitArg(defn.QuoteContextClass.typeRef, tree.span)
-    if (level == 0 && qctx.tpe.isInstanceOf[SearchFailureType])
-      ctx.error(missingArgMsg(qctx, defn.QuoteContextClass.typeRef, ""), ctx.source.atSpan(tree.span))
-
     tree.quoted match {
       case untpd.Splice(innerExpr) if tree.isTerm =>
         ctx.warning("Canceled splice directly inside a quote. '{ ${ XYZ } } is equivalent to XYZ.", tree.sourcePos)
-        typed(innerExpr, pt)
```

Comment:
We do not do the cancelation while typing because it optimized away some of the PCP check. Now we only cancel them during the `Staging` phase which also opitmizes away any synthesized quote and splice.